internal/controller/component_build_controller_pipeline.go

```diff
@@ -110,10 +110,26 @@ func (r *ComponentBuildReconciler) generatePaCPipelineRunConfigs(ctx context.Con
 		return nil, nil, err
 	}
 
+	// TODO delete the conditions after migration to the new Service Account done.
+	// If a Config Map with name "use-new-sa" exist, switch to the new Service Account.
+	useOldSA := true
+	if err := r.Client.Get(ctx, types.NamespacedName{Namespace: BuildServiceNamespaceName, Name: "use-new-sa"}, &corev1.ConfigMap{}); err == nil {
```

Comment:
I don't understand purpose of this CM

in reconcile you are ensuring that new SA exists, so why bother with this?

all new onboardings should use new one already

Reply:
This would allow us to perform migration when we are ready to do it, instead of migrating on operator deployment.